File: briareos-master/dev/project/briareos/zclient.py
from briareos.utils import loader
from briareos.common import *

from briareos.core import distributed_zmq


class ZClient:
    def __init__(self, client_config_path=ZCLIENT_CONFIG_PATH):
        self.config = loader.import_zclient_config(client_config_path)
        self.client = distributed_zmq.Client(self, self.config)

    def start(self):
        self.client.start()

    def stop(self):
        self.client.stop()

    def process(self, packet_payload, pipeline_name):
        return self.client.process(packet_payload, pipeline_name)

File: jbriareos-master/Tadeu_Freitas-jbriareos-c16b99c2c2d9/src/main/java/components/ZClient.java
package components;

import core.DistributedZMQ;

import java.util.UUID;

public class ZClient {
    DistributedZMQ.Client client;

    public ZClient(UUID id) {
        this.client = new DistributedZMQ.Client(id);
    }

    public void start() {
        this.client.start();
    }

    public void stop() {
        this.client.stop();
    }

    public void process(byte[] packetPayload, String pipelineName) {
        this.client.process(packetPayload, pipelineName);
    }
}
